# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1376610036 14400
#      Thu Aug 15 19:40:36 2013 -0400
# Node ID 12e5fa85f1bb7bae8b4e1106b86a39f00e14bfad
# Parent  533c980b5796be49571370aa87e04755ca01014d
Windows GUI: clear any terminal selection when pressing a key/pasting to terminal.

* libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
  (QConsolePrivate::sendConsoleText): clear selection before adding text to console.

diff --git a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
--- a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
+++ b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
@@ -1156,16 +1156,19 @@ void QConsolePrivate::startCommand (void
 
 void QConsolePrivate::sendConsoleText (const QString& s)
 {
   // Send the string in chunks of 512 characters. Each character is
   // translated into an equivalent keypress event.
 
 #define TEXT_CHUNK_SIZE 512
 
+  // clear any selection on inserting text
+  clearSelection();
+
   int len = s.length ();
   INPUT_RECORD events[TEXT_CHUNK_SIZE];
   DWORD nEvents = 0, written;
   HANDLE hStdIn = GetStdHandle (STD_INPUT_HANDLE);
 
   ZeroMemory (events, sizeof (events));
 
   for (int i = 0; i < len; i++)
